src/main/kotlin/notary/NotaryImpl.kt

```diff
@@ -79,10 +89,18 @@ class NotaryImpl(
     override fun onEthEvent(ethInputEvent: NotaryInputEvent.EthChainInputEvent): IrohaOrderedBatch {
         logger.info { "Notary performs ETH event" }
         return when (ethInputEvent) {
-            is NotaryInputEvent.EthChainInputEvent.OnEthSidechainDeposit -> onEthSidechainDeposit(ethInputEvent)
-
-        // TODO replace output with effective implementation
-            else -> IrohaOrderedBatch(arrayListOf())
+            is NotaryInputEvent.EthChainInputEvent.OnEthSidechainDeposit -> onEthSidechainDeposit(
+                ethInputEvent.hash,
+                ethInputEvent.user,
+                ethereumAssetId,
+                ethInputEvent.amount
+            )
+            is NotaryInputEvent.EthChainInputEvent.OnEthSidechainDepositToken -> onEthSidechainDeposit(
```

Comment:
Code duplication

Reply:
Different input events call function with different parameters (ethereumAssetId and ethInputEvent.token).